k9mail/src/main/java/com/fsck/k9/mailstore/ProvidedTempFileBody.java

```diff
@@ -82,11 +88,11 @@ public long getSize() {
         throw new IllegalStateException("Data must be fully written before it can be read!");
     }
 
-    public File getFile() throws IOException {
+    public Uri getProviderUri(String mimeType) throws IOException {
```

Comment:
`AttachmentInfoExtractor` already has a special case for `ProvidedTempFileBody`. You'd just do something like:

```java
File file = decryptedTempFileBody.getFile();    // Creates the file if necessary
uri = DecryptedFileProvider.getUriForProvidedFile(file, part.getMimeType());
```